version: "3.8"

services:
  seps:
    image: registry.gaeni.org/seps/ai:latest
    deploy:
      replicas: 1
      restart_policy:
        condition: on-failure
      labels:
        - "traefik.enable=true"
        - "traefik.docker.network=traefik-public"
        - "traefik.constraint-label=traefik-public"
        - "traefik.http.middlewares.https-redirect.redirectscheme.scheme=https"
        - "traefik.http.routers.seps.rule=Host(`seps-ai.gaeni.org`)"
        - "traefik.http.routers.seps.entrypoints=https"
        - "traefik.http.routers.seps.tls=true"
        - "traefik.http.routers.seps.tls.certresolver=le"
        - "traefik.http.services.seps.loadbalancer.server.port=8501"
    environment:
      OPENAI_API_KEY: "${OPENAI_API_KEY}"
      SERPER_API_KEY: "${SERPER_API_KEY}"
      GROQ_API_KEY: "${GROQ_API_KEY}"
      TAVILY_API_KEY: "${TAVILY_API_KEY}"
      OPENAI_MODEL: "gpt-4o"
      TEMPERATURE: "0.1"
      MAX_TOKENS: "2000"
    networks:
      - traefik-public

networks:
  traefik-public:
    external: true
